File: Laba5/c#/Laba5/Segm.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Laba5
{
    class Segm : Line
    {
        public Segm(int i, int j) : base(i, j) { }
        public override double Len()
        {
            return Math.Sqrt(Math.Pow(CoordX, 2) + Math.Pow(CoordY, 2));
        }
        public double Angle()
        {
            return Math.Acos(CoordX / Len()) * (180 / Math.PI);
        }
    }
}

File: Laba5/c++/Laba5.1/Segm.cpp
#include "Segm.h"

Segm::Segm(int x, int y) :Line(x, y) {}

double Segm:: Len()
{
	return sqrt(pow(CoordX, 2) + pow(CoordY, 2));
}

double Segm:: Angle()
{
    return acos(CoordX / Len()) * (180 / 3.14);
}

int Segm::x()
{
    return CoordX;
}
int Segm::y()
{
    return CoordY;
}
